src/libs/actions/IOU.js

```diff
@@ -75,6 +77,30 @@ function requestMoney(report, amount, currency, recipientEmail, participant, com
         iouReport = ReportUtils.buildOptimisticIOUReport(recipientEmail, debtorEmail, amount, chatReport.reportID, currency, preferredLocale);
     }
 
+    const optimisticTransaction = TransactionUtils.buildOptimisticTransaction(amount * 100, currency);
+    const optimisticTransactionData = {
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.TRANSACTION}${optimisticTransaction.transactionID}`,
+        value: optimisticTransaction,
+    };
+    const transactionSuccessData = {
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.TRANSACTION}${optimisticTransaction.transactionID}`,
+        value: {
+            pendingAction: null,
+        },
```

Comment:
Do we need to clear the errors here too?

Reply:
Good call

Reply:
Hmmm, well actually since there's no retry mechanism with existing RBR patterns, I don't think there's any way you would ever have success data applied on a transaction in Onyx which already had errors.

Reply:
So I think we're good here.